src/ir/effects.h

```diff
@@ -139,12 +142,12 @@ struct EffectAnalyzer
            throws;
   }
   bool hasSideEffects() const {
-    return hasGlobalSideEffects() || localsWritten.size() > 0 ||
-           transfersControlFlow() || implicitTrap || danglingPop;
+    return implicitTrap || localsWritten.size() > 0 || danglingPop ||
+           hasGlobalSideEffects() || transfersControlFlow();
   }
   bool hasAnything() const {
-    return hasSideEffects() || accessesLocal() || readsMemory ||
-           accessesGlobal() || isAtomic;
```

Comment:
Why reorder these? It looks like they were already roughly ordered from most to least likely, which is generally a good ordering.

Reply:
Yeah, I guess previous order makes more sense. Reverted